server/aesdsocket.c

```diff
@@ -0,0 +1,352 @@
+/****************************************************************************
+ *  Author:     Daniel Mendez
+ *  Course:     ECEN 5823
+ *  Project:    Assignment_5
+ *
+ ****************************************************************************/
+
+/**
+ * @file        aesdsocket.c
+ * @brief       Source file for socker server
+ *
+ * @details     This file contains the function to initialize the LETIMER0 module
+ *              with the following settings:
+ *
+ * @sources     - Beej Guide to Network Programming :https://beej.us/guide/bgnet/html/ Leveraged code from 6.1 A simple Stream Server with modifications
+ *              - Linux System Programming : Chapter 10 Signals Page 342
+ *
+
+ *
+ * @date        27 Sep 2023
+ * @version     1.0
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <arpa/inet.h>
+#include <sys/wait.h>
+#include <signal.h>
+#include <syslog.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <stdbool.h>
+
+
+#define PORT "9000"
+#define BUFFER_SIZE 1024
+#define BACKLOG 10   // how many pending connections queue will hold
+#define OUTPUT_FILE "/var/tmp/aesdsocketdata"
+#define ERROR_RESULT (-1)
+
+char * packet_buffer;
+
+static void signal_handler(int signo) {
+    printf("Signal Recieved %d \r\n",signo);
+    syslog(LOG_DEBUG, "Caught signal, exiting");
+    exit(EXIT_SUCCESS);
```

Comment:
@Silex93 how are you handling cleanup from your signal handler?  This is likely related to the valgrind warning messages.